ra-data-simple-rest 5.0.2 produces error when installing packages

<!-- Please do not submit support requests or "How to" questions here. For that, 
- go to Stack Overflow: https://stackoverflow.com/questions/tagged/react-admin),  or
  go to the react-admin discord server https://discord.gg/GeZF9sqh3N for community support or
- use the Professional Support (https://react-admin-ee.marmelab.com/#support) if you're an Enterprise Edition subscriber. -->

**What you were expecting:**
ra-data-simple-rest uses the same peer-dependency to ra-core as the other packages.

**What happened instead:**
```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR!
npm ERR! While resolving: ra-data-simple-rest@5.0.2
npm ERR! Found: ra-core@5.0.2
npm ERR! node_modules/ra-core
npm ERR!   ra-core@"^5.0.2" from ra-i18n-polyglot@5.0.2
npm ERR!   node_modules/ra-i18n-polyglot
npm ERR!     ra-i18n-polyglot@"^5.0.2" from react-admin@5.0.2
npm ERR!     node_modules/react-admin
npm ERR!       react-admin@"^5.0.2" from the root project
npm ERR!   ra-core@"^5.0.2" from ra-language-english@5.0.2
npm ERR!   node_modules/ra-language-english
npm ERR!     ra-language-english@"^5.0.2" from react-admin@5.0.2
npm ERR!     node_modules/react-admin
npm ERR!       react-admin@"^5.0.2" from the root project
npm ERR!   2 more (react-admin, ra-ui-materialui)
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer ra-core@"^4.0.0" from ra-data-simple-rest@5.0.2
npm ERR! node_modules/ra-data-simple-rest
npm ERR!   ra-data-simple-rest@"^5.0.2" from the root project
npm ERR!
npm ERR! Conflicting peer dependency: ra-core@4.16.19
npm ERR! node_modules/ra-core
npm ERR!   peer ra-core@"^4.0.0" from ra-data-simple-rest@5.0.2
npm ERR!   node_modules/ra-data-simple-rest
npm ERR!     ra-data-simple-rest@"^5.0.2" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

**Steps to reproduce:**
Use a package.json with my dependencies and run `npm i`

```
"dependencies": {
    "@emotion/react": "^11.11.4",
    "@emotion/styled": "^11.11.4",
    "@fontsource/roboto": "^5.0.13",
    "@mui/material": "^5.15.20",
    "ra-data-simple-rest": "^5.0.2",
    "react": "^18.3.1",
    "react-admin": "^5.0.2",
    "react-dom": "^18.3.1",
    "react-hook-form": "^7.52.0",
    "react-router-dom": "^6.2.1"
  }
```

**Other information:**
<!-- List any other information that is relevant to your issue. Stack traces, related issues, suggestions on how to fix, Stack Overflow links, forum links, etc. For visual or layout problems, please include images or animated gifs.-->

**Environment**

* React-admin version: 5.0.2
* Last version that did not exhibit the issue (if applicable): 4.16.18
* React version: 18.3.1
* Browser: Google Chrome

--- a/packages/ra-data-graphql-simple/package.json
+++ b/packages/ra-data-graphql-simple/package.json
@@ -41,7 +41,7 @@
     },
     "peerDependencies": {
         "graphql": "^15.6.0",
-        "ra-core": "^4.0.0"
+        "ra-core": "^5.0.0"
     },
     "devDependencies": {
         "cross-env": "^5.2.0",

--- a/packages/ra-data-graphql/package.json
+++ b/packages/ra-data-graphql/package.json
@@ -39,7 +39,7 @@
     },
     "peerDependencies": {
         "graphql": "^15.6.0 || ^16",
-        "ra-core": "^4.0.0"
+        "ra-core": "^5.0.0"
     },
     "devDependencies": {
         "cross-env": "^5.2.0",

--- a/packages/ra-data-simple-rest/package.json
+++ b/packages/ra-data-simple-rest/package.json
@@ -34,7 +34,7 @@
         "typescript": "^5.1.3"
     },
     "peerDependencies": {
-        "ra-core": "^4.0.0"
+        "ra-core": "^5.0.0"
     },
     "gitHead": "587df4c27bfcec4a756df4f95e5fc14728dfc0d7"
 }

--- a/packages/ra-input-rich-text/package.json
+++ b/packages/ra-input-rich-text/package.json
@@ -40,8 +40,8 @@
     "peerDependencies": {
         "@mui/icons-material": "^5.15.20",
         "@mui/material": "^5.15.20",
-        "ra-core": "^4.0.0",
-        "ra-ui-materialui": "^4.0.0",
+        "ra-core": "^5.0.0",
+        "ra-ui-materialui": "^5.0.0",
         "react": "^18.0.0 || ^19.0.0",
         "react-dom": "^18.0.0 || ^19.0.0"
     },

